test/unit/streaming-client.spec.ts

```diff
@@ -113,42 +113,40 @@ describe('streaming-client', () => {
         });
 
         it('adds event listener to httpResponse', () => {
-            // Setup
+            // Arrange
             const res = sut.start();
             let statusCode = null;
-            sut.on('httpResponse', code => {
-                statusCode = code;
-            });
+            sut.on('httpResponse', code => statusCode = code);
 
             // Act
             mockClient.emit('httpResponse', { statusCode: 401 });
 
             // Assert
             expect(statusCode).toBe(401);
             expect(res.writable).toBe(false);
-            expect(() => { res.write('message'); }).toThrow();
+            res.on('error', (err: any) => expect(err.code).toBe('ERR_STREAM_WRITE_AFTER_END'));
+            res.write('messsage');
```

Comment:
The original fails unit tests for the new Node versions. It's because `res` doesn't actually throw but emits an error because it extends from `EventEmitter`, so to handle the end of stream error, we provide a listener and add an assertion in there.